src/components/Todo.vue

```diff
@@ -0,0 +1,150 @@
+<template>
+  <div class="todo">
+    <template class="todo-intarface">
+      <h1
+        :class="{
+          red: checkCountOfDoneTask() < 0.5,
+          yellow: checkCountOfDoneTask() >= 0.5 && checkCountOfDoneTask() < 1,
+          green: checkCountOfDoneTask() === 1,
+        }"
+      >
+        TODO LIST
+      </h1>
+      <input
+        class="todo-input"
+        v-on:keydown.enter="addTodo"
+        type="text"
+        v-model="todo"
+      />
+      <button class="todo-btn" @click="addTodo">add new task</button>
+    </template>
+    <ul class="todo-list">
+      <li
+        v-for="(t, idx) of todos"
+        :key="idx"
+        class="todo-item"
+        @click.self="removeTodo(idx)"
+      >
+        <input
+          class="todo-checkbox"
+          @change.self="changeStatus(t)"
+          type="checkbox"
+          name="is-done"
+        />
+        <span :class="{ done: t.status }" @click="removeTodo(idx)"
+          >{{ `${idx + 1}.` }} {{ t.title }}</span
+        >
+      </li>
+    </ul>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "ToDo",
+  data() {
+    return {
+      todos: [],
+      todo: "",
+    };
+  },
+  methods: {
+    addTodo() {
+      if (this.todo) {
+        this.todos.push({
+          title: this.todo,
+          status: false,
+        });
+      }
+      this.todo = "";
+    },
+    changeStatus(task) {
+      task.status = !task.status;
+    },
+    removeTodo(idx) {
+      this.todos = this.todos.filter((el, id) => id !== idx);
+    },
+    checkCountOfDoneTask() {
```

Comment:
This method works, but it's a perfect candidate to be replaced with a `computed()`.
Thant way it will only be calculated once and it will be cached.

https://vuejs.org/guide/essentials/computed.html

Reply:
Unresolving this:

- [ ] Move `checkCountOfDoneTask` to `computed`.
- [ ] When using it, do it with `checkCountOfDoneTask` instead of `checkCountOfDoneTask()` (because after you move it it is no longer a method!)
- [ ] `removeTask` shouldn't be a `computed`. `removeTask` is a perfectly good method.

Reply:
In previous time I misunderstood which method I should to put on in computed property and mistakenly tried to convert another method to a computed property. This time I hope did it properly.